packages/pricing/src/models/price-set.ts

```diff
@@ -1,52 +1,74 @@
-import { generateEntityId } from "@medusajs/utils"
+import {
+  DALUtils,
+  createPsqlIndexStatementHelper,
+  generateEntityId,
+} from "@medusajs/utils"
 import {
   BeforeCreate,
   Cascade,
   Collection,
   Entity,
+  Filter,
   ManyToMany,
-  OneToMany,
   OnInit,
-  OptionalProps,
+  OneToMany,
   PrimaryKey,
+  Property,
 } from "@mikro-orm/core"
-
-import MoneyAmount from "./money-amount"
 import PriceRule from "./price-rule"
 import PriceSetMoneyAmount from "./price-set-money-amount"
 import PriceSetRuleType from "./price-set-rule-type"
 import RuleType from "./rule-type"
 
-@Entity()
-export default class PriceSet {
-  [OptionalProps]?: "price_set_money_amounts" | "rule_types" | "money_amounts"
+const tableName = "price_set"
+const PriceSetDeletedAtIndex = createPsqlIndexStatementHelper({
+  tableName: tableName,
+  columns: "deleted_at",
+  where: "deleted_at IS NOT NULL",
+})
 
+@Entity({ tableName })
+@Filter(DALUtils.mikroOrmSoftDeletableFilterOptions)
+export default class PriceSet {
   @PrimaryKey({ columnType: "text" })
   id!: string
 
   @OneToMany(() => PriceSetMoneyAmount, (psma) => psma.price_set, {
-    cascade: [Cascade.REMOVE],
+    cascade: ["soft-remove" as Cascade],
```

Comment:
suggestion: I would suggest to add persist in all custom cascade like this, I know that if it not present it can create issues, something we encountered with the product

Reply:
same for all oneToMany if we specify cascade

Reply:
are the issues related to soft deletes / restoring?